### PR TITLE
Fix invalid handling of possible null values

### PR DESCRIPTION
**Pull request by mxr576**
Developer category reference can be null on developer category-specific rate plans.

Steps to reproduce:
1. Create a developer category and a rate plan for it.
2. Delete the developer category
3. Open a page that uses this code, the developer category becomes missing (null) on the developer category-specific rate plan therefore it cannot be loaded.

### CODE REVIEW

File: src/Entity/Access/RatePlanAccessControlHandler.php [hunk 1]
```diff
@@ -84,11 +84,21 @@ protected function checkAccess(EntityInterface $entity, $operation, AccountInter
     // does not belong to rate_plan category.
     /** @var \Drupal\apigee_edge\Entity\DeveloperInterface $developer */
     if ($rate_plan instanceof DeveloperCategoryRatePlanInterface) {
-      if (($category = $rate_plan->getDeveloperCategory()) && ($developer = $this->entityTypeManager->getStorage('developer')->load($account->getEmail()))) {
```
Comment (mxr576) [inline src/Entity/Access/RatePlanAccessControlHandler.php:87]: Variable assignment in a condition is dangerous and should be avoided.

**Comment by mxr576** [comment]
Review together with #291

**Comment by mxr576** [comment]
Hi, can somebody review this change?

**Comment by raakesh-blokhra** [comment]
Hi @mxr576 Can you link this and #291 to relevant issues and also if possible, please provide steps for testing. The code looks good, would like to test and then review. Thanks

**Comment by mxr576** [comment]
Hi @raakesh-blokhra,

I do not have perms to add issue references as far as I see.

(Updated the issue description based on my memories)

**Comment by shishir-intelli** [comment]
Closed via https://github.com/apigee/apigee-m10n-drupal/pull/393

**Pull request closed without merging**